src/Node.ts

```diff
@@ -1,21 +1,37 @@
 import { fabric } from "fabric";
 import { Pointer } from "./Pointer";
 import { Point } from "./Types";
+import Config = require("./Config");
 
 export abstract class Node {
     public data: number;
     public next: Pointer
     protected representation: fabric.Group;
 
-    constructor(data: number, canvas: fabric.Canvas) {
+    constructor(data: number, canvas: fabric.Canvas, pointer: Pointer) {
```

Comment:
This whole thing looks great, my one minor nitpick is that I would rather just pass in the coordinates here than the entire pointer object, just so that its clear which pieces of information the new Node does and doesn't need to have. 

I'll go ahead and merge this in and make that small change myself. 

Thanks a bunch!